### ISSUE TITLE
Flutter Image Loading 

### ISSUE THREAD

**Issue by AseemWangoo** [flutter/flutter#21670]
Hi all,

I am trying to load n number of images (from network) in Grid View with the pagination concept. When I scroll, the app crashes. 

I also tried to load images from my gallery in Grid View, the loading of images takes time and makes the app to crash in this case too.

May I get some tips on achieving this task ?

Code snippet
---------------------------------
return GridView.builder(
            controller: scrollController,
            itemCount: postList['edges'].length,
            physics: const AlwaysScrollableScrollPhysics(),
            gridDelegate: new SliverGridDelegateWithFixedCrossAxisCount(
                crossAxisCount: 3),
            itemBuilder: (BuildContext context, int index) {
              return new GestureDetector(
                  child: new Card(
                    elevation: 0.0,
                    margin: const EdgeInsets.all(2.0),
                    shape: new RoundedRectangleBorder(
                        borderRadius:
                            new BorderRadius.all(new Radius.circular(0.0))),
                    child: new Image.network(
                      url,
                      fit: BoxFit.cover,
                      scale: 10.0,     // i don't think whether this is even useful....
                    ),                    
                  ),                  
                );
            },
          );
                  },
                );
            },
          );

**Comment by zoechi** [comment]
Could be because it runs out of memory.
Can you please add the output of `flutter doctor -v`
and if possible some error log output.

**Comment by AseemWangoo** [comment]
Yes it runs out of memory. While debugging, I always come up with the following line

/libc    (20866): Fatal signal 11 (SIGSEGV), code 128, fault addr 0x0 in tid 20885 (1.ui)
Lost connection to device.
Exited (sigterm)

**Comment by AseemWangoo** [comment]
[√] Flutter (Channel beta, v0.4.4, on Microsoft Windows [Version 10.0.16299.431], locale en-US)
    • Flutter version 0.4.4 at C:\Users\AseemW\Downloads\flutter_windows_v0.4.4-beta\flutter
    • Framework revision f9bb4289e9 (4 months ago), 2018-05-11 21:44:54 -0700
    • Engine revision 06afdfe54e
    • Dart version 2.0.0-dev.54.0.flutter-46ab040e58

[!] Android toolchain - develop for Android devices (Android SDK 27.0.3)
    • Android SDK at C:\Users\AseemW\AppData\Local\Android\sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-27, build-tools 27.0.3
    • Java binary at: C:\Old Win7\D\Softwares\android-studio-ide-181.4773949-windows\android-studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b03)
    ! Some Android licenses not accepted.  To resolve this, run: flutter doctor --android-licenses

[√] Android Studio (version 2.3)
    • Android Studio at C:\Program Files\Android\Android Studio
    X Flutter plugin not installed; this adds Flutter specific functionality.
    X Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_112-release-b06)

[√] Android Studio (version 3.2)
    • Android Studio at C:\Old Win7\D\Softwares\android-studio-ide-181.4773949-windows\android-studio
    X Flutter plugin not installed; this adds Flutter specific functionality.
    X Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b03)

[!] Connected devices
    ! No devices available

! Doctor found issues in 2 categories.

**Comment by zoechi** [comment]
That's quite an old Flutter version.
Can you reproduce in a recent `beta` version?

**Comment by AseemWangoo** [comment]
thanks for noticing
